### PR TITLE
Removed getRequest calls

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | is a part of #2106 
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR removes the deprecated `$this->getRequest()` calls from controller.

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@danrot the removed functions want be used anymore. should we add them to the upgrade?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@ CHANGELOG for Sulu
 ==================
 
 * dev-develop
+    * ENHANCEMENT #3080 [ALL]                 Removed getRequest calls
```
Comment (danrot) [inline CHANGELOG.md]: It's `All`, not `ALL`

**Comment by danrot** [comment]
We should at least check with @turbo-ele if we can remove the methods from the `RestController` that easily, or if they are still using them in some projects.

**Comment by turbo-ele** [comment]
@danrot @wachterjohannes i've just checked our biggest project and haven't seen any uses of the functions in RestController that have been removed.
I guess it should be safe to remove these functions

**Comment by danrot** [comment]
@wachterjohannes Let's still add an upgrade notice for the deleted methods.

**Comment by wachterjohannes** [comment]
@danrot upgrade added

**Pull request merged**